### PR TITLE
Mag calibration fix

### LINKED ISSUE #1188: BUG - MAG L1B Calibration matrix
### Description of the issue

Calibration matrix is outputting the wrong values, the matrix multiplication needs to be swapped. 

### Steps to reproduce the issue

_No response_

### Expected behavior (What should happen)

Per email: 
"When I multiply the calibration matrix for MAGo in range 0 with the unity vector [1, 1, 1]T, I get the following vector: [2.2972, 2.2415, 2.2381]T"

### Actual behavior (What does happen)

_No response_

### Code Snippet:

```python
Code
```

### Additional notes, affected areas, and suggested fixes

_No response_

### PR DESCRIPTION
**Pull request by maxinelasp**
# Change Summary
Swapped the order of matrix multiplication and added some extra tests.

Test numbers now come primarily from MAG team instead of my own math.

Fixes #1188

### CODE REVIEW

**Review by mfacchinelli — APPROVED** [review]
Looks great, thanks Maxine!

**Review by vmartinez-cu — APPROVED** [review]
LGTM

File: imap_processing/tests/mag/test_mag_l1b.py [hunk 1]
```diff
@@ -132,3 +136,33 @@ def test_mag_compression_scale(mag_l1a_dataset):
     # width of 14 should be multiplied by 4
     scaled_vectors = calibrated_vectors * 4
     assert np.allclose(output["vectors"].data[3][:3], scaled_vectors)
+
+
+def test_calibrate_vector():
+    # from MFOTOURFO
```
Comment (tech3371) [inline imap_processing/tests/mag/test_mag_l1b.py:142]: What does this mean? Just curious. Can you add comment for future reference?

Comment (maxinelasp) [reply imap_processing/tests/mag/test_mag_l1b.py:142]: Oh, that's the name of the variable in the calibration CDF file

**Comment by nkerman** [comment]
LGTM!

File: imap_processing/tests/mag/test_mag_l1b.py (hunk 1 above)
Comment (mfacchinelli) [reply imap_processing/tests/mag/test_mag_l1b.py:142]: It stands for Measurement Frame Outboard to Universal Reference Frame Outboard. It represents the conversion matrix from the 2 reference frames.

Comment (tech3371) [reply imap_processing/tests/mag/test_mag_l1b.py:142]: It would have been helpful to add that. A year from now, this clarification will be helpful

**Pull request merged**